tests/cupyx_tests/scipy_tests/interpolate_tests/test_polyint.py

```diff
@@ -0,0 +1,236 @@
+            pytest.skip()
+        true_poly = xp.poly1d([-2, 3, 1, 5, -4])
+        test_xs = xp.linspace(-5, 5, 5, dtype=dtype)
+        xs = xp.linspace(-1, 1, 5, dtype=dtype)
+        ys = true_poly(xs)
+        P = scp.interpolate.KroghInterpolator(xs, ys)
+        D = P.derivatives(test_xs)
+        return D
+
+    @testing.for_all_dtypes(no_bool=True, no_complex=True)
+    @testing.numpy_cupy_allclose(scipy_name='scp', atol=1e-7, rtol=1e-7)
+    def test_low_derivatives(self, xp, scp, dtype):
+        if xp.dtype(dtype).kind in 'u':
+            pytest.skip()
+        true_poly = xp.poly1d([-2, 3, 1, 5, -4])
+        test_xs = xp.linspace(-5, 5, 5, dtype=dtype)
+        xs = xp.linspace(-1, 1, 5, dtype=dtype)
+        ys = true_poly(xs)
+        P = scp.interpolate.KroghInterpolator(xs, ys)
+        D = P.derivatives(test_xs, len(xs) + 2)
+        return D
+
+    @testing.for_all_dtypes(no_bool=True, no_complex=True)
+    @testing.numpy_cupy_allclose(scipy_name='scp', atol=1e-7, rtol=1e-7)
+    def test_derivative(self, xp, scp, dtype):
+        if xp.dtype(dtype).kind in 'u':
+            pytest.skip()
+        true_poly = xp.poly1d([-2, 3, 1, 5, -4])
+        test_xs = xp.linspace(-5, 5, 5, dtype=dtype)
+        xs = xp.linspace(-1, 1, 5, dtype=dtype)
+        ys = true_poly(xs)
+        P = scp.interpolate.KroghInterpolator(xs, ys)
+        m = 10
+        for i in range(m):
+            return P.derivative(test_xs, i)
+
+    @testing.for_all_dtypes(no_bool=True, no_complex=True)
+    @testing.numpy_cupy_allclose(scipy_name='scp', atol=1e-7, rtol=1e-7)
+    def test_high_derivative(self, xp, scp, dtype):
+        if xp.dtype(dtype).kind in 'u':
+            pytest.skip()
+        true_poly = xp.poly1d([-2, 3, 1, 5, -4])
+        test_xs = xp.linspace(-5, 5, 5, dtype=dtype)
+        xs = xp.linspace(-1, 1, 5, dtype=dtype)
+        ys = true_poly(xs)
+        P = scp.interpolate.KroghInterpolator(xs, ys)
+        D = P.derivative(test_xs, 2 * len(xs))
+        return D
+
+    @testing.for_all_dtypes(no_bool=True, no_complex=True)
+    @testing.numpy_cupy_allclose(scipy_name='scp', atol=1e-7, rtol=1e-7)
+    def test_hermite(self, xp, scp, dtype):
+        if xp.dtype(dtype).kind in 'u':
+            pytest.skip()
+        true_poly = xp.poly1d([-2, 3, 1, 5, -4])
+        test_xs = xp.linspace(-5, 5, 5, dtype=dtype)
+        xs = xp.linspace(-1, 1, 5, dtype=dtype)
+        ys = true_poly(xs)
+        P = scp.interpolate.KroghInterpolator(xs, ys)
+        D = P(test_xs)
+        return D
+
+    @pytest.mark.parametrize('n', [0, [0], [0, 1]])
+    @testing.for_all_dtypes(no_bool=True, no_complex=True)
+    @testing.numpy_cupy_allclose(scipy_name='scp', atol=1e-7, rtol=1e-7)
+    def test_hermite_derivative1(self, xp, scp, dtype, n):
+        if xp.dtype(dtype).kind in 'u':
+            pytest.skip()
+        true_poly = xp.poly1d([-2, 3, 1, 5, -4])
+        xs = xp.linspace(-1, 1, 5, dtype=dtype)
+        ys = true_poly(xs)
```

Comment:
Following https://github.com/cupy/cupy/pull/6990#discussion_r954856930. To construct the polynomial with some derivative value, you should change `xs` and `ys` to pass it. 

https://docs.scipy.org/doc/scipy/reference/generated/scipy.interpolate.KroghInterpolator.html
> yi : array_like
>     Known y-coordinates. When an xi occurs two or more times in a row, the corresponding yi’s represent derivative values.

Reply:
Hi, I understand this and improved my impl pointing to the 0th derivative (xi) at positions 1st, 2nd, and 3rd.
```python
def func():
    xs = xp.array([0, 0, 0], dtype=dtype)
    ys = xp.array([1, 2, 3], dtype=dtype)
    P = scp.interpolate.KroghInterpolator(xs, ys)
    return P.derivatives(0)
```
I think this is the same one you meant!?

Maybe we'll have to change a bit in the `_Interpolator` class. Currently getting:
<details>
<summary>error log</summary>

<p>

```python
E           AssertionError: Only cupy raises error
E           
E           Traceback (most recent call last):
E             File "/home/khushi/Documents/cupy/cupy/testing/_loops.py", line 45, in _call_func
E               result = impl(*args, **kw)
E             File "/home/khushi/Documents/cupy/tests/cupyx_tests/scipy_tests/interpolate_tests/test_polyint.py", line 119, in test_hermite_derivative1
E               return P.derivatives(0)
E             File "/home/khushi/Documents/cupy/cupyx/scipy/interpolate/_polyint.py", line 160, in derivatives
E               x, x_shape = self._prepare_x(x)
E             File "/home/khushi/Documents/cupy/cupyx/scipy/interpolate/_polyint.py", line 81, in _prepare_x
E               x = _asarray_validated(x, check_finite=False, as_inexact=True)
E             File "/home/khushi/Documents/cupy/cupyx/scipy/_lib/_util.py", line 69, in _asarray_validated
E               if not cupy.issubdtype(a, cupy.inexact):
E             File "/home/khushi/anaconda3/envs/cupy-dev/lib/python3.10/site-packages/numpy/core/numerictypes.py", line 416, in issubdtype
E               arg1 = dtype(arg1).type
E           TypeError: Cannot interpret '0' as a data type
```

</p>
</details>

Reply:
The values of `xs` and `ys` are okay. (btw, you're setting 0th, 1st, and 2nd derivatives at the point x = 0)

I think testing evaluated values at points given by `linspace` is better as your `test_hermite` does, instead of derivatives at a single point that is the same as the known one.

```py
def func():
    xs = xp.array([0, 0, 0], dtype=dtype)
    ys = xp.array([1, 2, 3], dtype=dtype)
    P = scp.interpolate.KroghInterpolator(xs, ys)
    test_xs = xp.linspace(-5, 5, 5, dtype=dtype)
    D = P(test_xs)
    return D
```

Reply:
(The error you saw is because of that you passed a Python scalar)

Reply:
Thanks, updated.